### PR TITLE
Upgrade tooling and dependencies

### PR DESCRIPTION
**Pull request by eguzki**
### what

* bumped Go to 1.17
* bumped k8s deps to 0.22.1
* bumped client-go to 0.22.1
* install tooling using `go install` ([Deprecation of 'go get' for installing executables](https://go.dev/doc/go-get-install-deprecation))
* kustomize v4.5.7
* controller-gen v0.9.2
* github.com/RHsyseng/operator-utils v1.4.9

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,27 +1,89 @@
 module github.com/3scale/apicast-operator
 
-go 1.13
+go 1.17
 
 require (
 	github.com/RHsyseng/operator-utils v0.0.0-20200213165520-1a022eb07a43
```
Comment (MStokluska) [inline go.mod:6]: Think we can bump this as well to 	github.com/RHsyseng/operator-utils v1.4.9 to stay in line with what's coming to 3scale operator?

**Comment by MStokluska** [comment]
@eguzki changes look good, just have a question about rhsyseng dep and it looks good to merge to me

File: go.mod (hunk 1 above)
Comment (eguzki) [reply go.mod]: done

**Review by MStokluska — APPROVED** [review]
looks good to me

**Pull request merged**